### PR TITLE
Support FX Mixer for sample tracks and add controls to sample track window

### LINKED ISSUE #3268: Sample Track - Effect Chain Window - Add Title and Remove The Fullscreen Icon
Means nothing to me, but something I noticed.

If you click on a "sample track"s label it will open up an "effects chain" window for it.
1.) Might be a good idea to add the sample tracks name as a title for this effects window to know what sample track it is linked to.  (Might be tough due to the width of the effects chain, probably why it is not there)
2.) This "effects chain" window also contains a maximize/fullscreen icon to the left of its "x" close icon in the titlebar, which should probably be removed.

### PR DESCRIPTION
**Pull request by PhysSong**
Continuation from @grejppi's original work #3632.
Before merging, I need to fix indentation and add some code for loading/saving window states. See https://github.com/LMMS/lmms/pull/3632#issuecomment-317638049 and https://github.com/LMMS/lmms/pull/3632#pullrequestreview-54781294.
Closes #3268.

### CODE REVIEW

**Comment by PhysSong** [comment]
Since the original PR was approved, I'd like to merge this after a few days if no one objects.

**Comment by PhysSong** [comment]
I was busy to take care of this PR. I'll merge it after reducing duplications soon.

**Comment by PhysSong** [comment]
I found some bugs with the FX spin box. I need to fix that first before merging.

File: src/tracks/SampleTrack.cpp [hunk 1]
```diff
@@ -562,10 +565,12 @@ SampleTrack::SampleTrack( TrackContainer* tc ) :
 							tr( "Volume" ) ),
 	m_panningModel( DefaultPanning, PanningLeft, PanningRight, 0.1f,
 					this, tr( "Panning" ) ),
+	m_effectChannelModel( 0, 0, 0, this, tr( "FX channel" ) ),
 	m_audioPort( tr( "Sample track" ), true, &m_volumeModel, &m_panningModel, &m_mutedModel )
 {
 	setName( tr( "Sample track" ) );
 	m_panningModel.setCenterValue( DefaultPanning );
```
Comment (karmux) [inline src/tracks/SampleTrack.cpp]: After this line add `m_effectChannelModel.setRange( 0, Engine::fxMixer()->numChannels()-1, 1);`.
This fixes problem with changing FX channel for newly added sample tracks.

**Comment by karmux** [comment]
Add missing line and merge?

**Comment by karmux** [comment]
@PhysSong add missing line and merge?

**Comment by PhysSong** [comment]
Fixed remaining bugs and rebased commits. You can find the version before rebasing here: https://github.com/PhysSong/lmms/tree/sample-track-fx-bak

**Comment by zonkmachine** [comment]
The top post still mentions things needed to be done here but I assume that it has just not been updated.
Works like a charm. This should be merged.

**Comment by zonkmachine** [comment]
@PhysSong Merge?

**Comment by PhysSong** [comment]
Sorry, I almost forgot this one. I'll self-review this one soon.

**Pull request merged**